src/Client.php

```diff
@@ -190,13 +190,10 @@ private function configureDefaults(array $config)
         // Add the default user-agent header.
         if (!isset($this->config['headers'])) {
             $this->config['headers'] = ['User-Agent' => default_user_agent()];
-        } else {
-            // Add the User-Agent header if one was not already set.
-            foreach (array_keys($this->config['headers']) as $name) {
-                if (strtolower($name) === 'user-agent') {
-                    return;
-                }
-            }
+        }
+
+        $lowercasedHeaders = array_change_key_case($this->config['headers'], CASE_LOWER);
+        if (!isset($lowercasedHeaders['user-agent'])) {
```

Comment:
I would still leave this in an else branch. This now runs nevertheless of the previous if, while it should only run if the previous condition is false.